servers/src/main/java/tachyon/worker/block/allocator/RRAllocator.java

```diff
@@ -0,0 +1,97 @@
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block.allocator;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.common.base.Preconditions;
+
+import tachyon.worker.block.BlockStoreLocation;
+import tachyon.worker.block.BlockMetadataManager;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.StorageTier;
+import tachyon.worker.block.meta.TempBlockMeta;
+
+/**
+ * A Round robin allocator that allocates a block in the storage dir.
+ */
+public class RRAllocator implements Allocator {
+  private final BlockMetadataManager mMetaManager;
+
+  // We need to remember the last dir index for every storage tier
+  private Map<StorageTier, Integer> mTierDirs = new HashMap<StorageTier, Integer>();
+
+  public RRAllocator(BlockMetadataManager metadata) {
+    mMetaManager = Preconditions.checkNotNull(metadata);
+    
+    for (StorageTier tier : mMetaManager.getTiers()) {
+      mTierDirs.put(tier, -1);
+    }
+  }
+
+  @Override
+  public TempBlockMeta allocateBlock(long userId, long blockId, long blockSize,
+      BlockStoreLocation location) throws IOException {
+
+    if (location.equals(BlockStoreLocation.anyTier())) {
+      int tierAlias = 0; //always starting from the first tier
+      for (int i = 0; i < mMetaManager.getTiers().size(); i ++) {
+        StorageTier tier = mMetaManager.getTiers().get(tierAlias);
+        int dirIndex = getNextDirInTier(tier, blockSize);
+        if (dirIndex >= 0) {
+          mTierDirs.put(tier, dirIndex); // update
+          return new TempBlockMeta(userId, blockId, blockSize, tier.getDir(dirIndex));
+        } else { // we didn't find one in this tier, go to next tier
+          tierAlias = (tierAlias + 1) % mMetaManager.getTiers().size();
+        }
+      }
+    } else if (location.equals(BlockStoreLocation.anyDirInTier(location.tierAlias()))) {
+      StorageTier tier = mMetaManager.getTier(location.tierAlias());
+      int dirIndex = getNextDirInTier(tier, blockSize);
+      if (dirIndex >= 0) {
+        mTierDirs.put(tier, dirIndex); // update
+        return new TempBlockMeta(userId, blockId, blockSize, tier.getDir(dirIndex));
+      }
+    } else {
+      StorageTier tier = mMetaManager.getTier(location.tierAlias());
+      StorageDir dir = tier.getDir(location.dir());
+      if (dir.getAvailableBytes() >= blockSize) {
+        return new TempBlockMeta(userId, blockId, blockSize, dir);
+      }
```

Comment:
In this case, do we need to account for the fact the dir was allocated to? It would be good to document if the allocator only considers non specific writes in its RR policy?